src/main/zapHomeFiles/hud/libs/event-bus.js

```diff
@@ -0,0 +1,2 @@
+import Vue from 'vue';
+export const EventBus = new Vue();
```

Comment:
love this shared module

Reply:
Because the events are registered in this external bus, they all need to be individually torn down:
e.g.
```
	beforeDestroy () {
		eventBus.$off('showSiteTreeModal')
	}
```

You could create a global mixin here, but I think that the difficulty you would face is that you will have to contort yourself to know which component is being destroyed. Probably better to tear them down in the individual components, because that way later contributors can SEE that they are being unregistered.